core/src/main/java/org/apache/calcite/adapter/jdbc/JdbcRules.java

```diff
@@ -262,6 +263,17 @@ private JdbcRules() {
       this(clazz, (Predicate<R>) predicate, in, out, relBuilderFactory,
           description);
     }
+
+    /**
+     * Creates name for child JdbcConverter rules in format:
+     * RuleName(out:CONVENTION_NAME)
+     *
+     * @param out output convention
+     * @return rule name
+     */
+    protected static String getRuleName(Class<?> clazz, JdbcConvention out) {
+      return String.format(Locale.ROOT, "%s(out:%s)", clazz.getSimpleName(), out);
+    }
```

Comment:
Just one question, can we make more effort to move this method to super class ConverterRule ? It seems that this is a common pattern, it would be nice if we can unify the naming rules so there is less change we encounter the similar problem.

Reply:
This pattern was chosen by me, because it's similar to the one in ```ConverterRule``` constructor: 
```
        description == null
            ? "ConverterRule(in:" + in + ",out:" + out + ")"
            : description)
```
Only difference in my code is that I didn't use **in** rel trait argument, since for jdbc rules it's the same everywhere. I can move the method to ```ConverterRule```, but then ignoring ```in``` argument won't be good I think. So, should I move the method up in the hierarchy with pattern like ```RuleName(in:" + in + ",out:" + out + ")"``` ?

Reply:
The problem only exists for JdbcConvention, right? Other conventions are singletons. If so, I would be OK if this code stayed in JdbcRules.

Reply:
Even though other conventions are singletons, we didn't use it as part of the description, i.e. the ElasticSearchConverter rules [1], i believe they have the same issue.

[1] https://github.com/apache/calcite/blob/22577e488e04d4056e15af96bac4246fd810f0fa/elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchRules.java#L240

Reply:
Well, exactly. If there is only one instance of convention C, you can only have one instance of a rule that uses C. But if there are two instances of the rule that use c1 and c2, you have to include c1 and c2 in the name of those rules so that their names are different.

Maybe ElasticSearch convention should NOT be a singleton. I suppose we'll run into the bug when someone wants to run a federated query across more than one ElasticSearch database.

Reply:
I checked again and almost every adapter converter rules do not include the convention in their description, maybe we can alway include the convention into the rule name and unify the logic into `ConverterRule`.

Reply:
@danny0405 , @julianhyde  please, take a look again whether I correctly understood idea. All converter rules description now will contain **in,out** rel traits.